eclair-core/src/main/scala/fr/acinq/eclair/db/pg/PgChannelsDb.scala

```diff
@@ -40,27 +40,29 @@ class PgChannelsDb(implicit ds: DataSource, lock: PgLock) extends ChannelsDb wit
   val DB_NAME = "channels"
   val CURRENT_VERSION = 3
 
-  def migration23(statement: Statement): Unit = {
-    statement.executeUpdate("ALTER TABLE local_channels ADD COLUMN created_timestamp BIGINT")
-    statement.executeUpdate("ALTER TABLE local_channels ADD COLUMN last_payment_sent_timestamp BIGINT")
-    statement.executeUpdate("ALTER TABLE local_channels ADD COLUMN last_payment_received_timestamp BIGINT")
-    statement.executeUpdate("ALTER TABLE local_channels ADD COLUMN last_connected_timestamp BIGINT")
-    statement.executeUpdate("ALTER TABLE local_channels ADD COLUMN closed_timestamp BIGINT")
-  }
-
   inTransaction { pg =>
     using(pg.createStatement()) { statement =>
-      getVersion(statement, DB_NAME, CURRENT_VERSION) match {
-        case 2 =>
-          logger.warn(s"migrating db $DB_NAME, found version=2 current=$CURRENT_VERSION")
+
+      def migration23(statement: Statement): Unit = {
+        statement.executeUpdate("ALTER TABLE local_channels ADD COLUMN created_timestamp BIGINT")
+        statement.executeUpdate("ALTER TABLE local_channels ADD COLUMN last_payment_sent_timestamp BIGINT")
+        statement.executeUpdate("ALTER TABLE local_channels ADD COLUMN last_payment_received_timestamp BIGINT")
+        statement.executeUpdate("ALTER TABLE local_channels ADD COLUMN last_connected_timestamp BIGINT")
+        statement.executeUpdate("ALTER TABLE local_channels ADD COLUMN closed_timestamp BIGINT")
+      }
+
+      getVersion(statement, DB_NAME) match {
```

Comment:
nit: the order of the `case` is a bit hard to read, what about using a consistent order in all files?
What makes the most sense to me is:

- list the supported previous versions (`case Some(1)`, `case Some(2)`, etc)
- then the current version (`case Some(CURRENT_VERSION)`)
- then catch unsupported versions (`case Some(unknownVersion)`)
- then `case None`

Reply:
Nice catch! It's an oversight, I actually reordered all of them except that one. Fixed with 8e7e76fbd.

The only difference between your ordering and mine is that I put `None` first. The rationale is that it makes sense to have the full from-scratch init first, because it's what describes the current schema of the database.

Reply:
That makes sense, SGTM.